dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileTiny.cpp

```diff
@@ -178,13 +225,13 @@ ColumnFilePersistedPtr ColumnFileTiny::restoreFromCheckpoint(
 
     // Write index data page to local ps
     auto new_index_infos = std::make_shared<IndexInfos>();
-    for (const auto & index : *index_infos)
+    for (const auto & index_pb : *index_infos)
     {
-        auto new_index_page_id = put_remote_page(index.index_page_id);
-        if (index.vector_index)
-            new_index_infos->emplace_back(new_index_page_id, index.vector_index);
-        else
-            new_index_infos->emplace_back(new_index_page_id, std::nullopt);
+        details::integrityCheckIndexInfoV2(index_pb);
+        auto new_index_page_id = put_remote_page(index_pb.index_page_id());
+        auto new_index_pb = index_pb;
+        new_index_pb.set_index_page_id(new_index_page_id);
+        new_index_infos->emplace_back(std::move(index_pb));
```

Comment:
```suggestion
        new_index_infos->emplace_back(std::move(new_index_pb));
```

typo?

Reply:
Thanks! I will fix it and test with disaggregate mode later.

Reply:
Disaggregate mode looks fine and this is a FAP path. I will checkout this PR in FAP mode later.